sdk/python/feast/data_source.py

```diff
@@ -437,74 +405,211 @@ def date_partition_column(self, date_partition_column):
     @classmethod
     def from_proto(cls, data_source_proto: DataSourceProto):
         """
-        Creates a DataSource from a protobuf representation of an data source
+        Creates a DataSource from a protobuf representation of a data source
+        """
+        raise NotImplementedError
 
-        Args:
-            data_source_proto: A protobuf representation of a DataSource
+    def to_proto(self) -> DataSourceProto:
+        """
+        Converts an DataSourceProto object to its protobuf representation.
+        """
+        raise NotImplementedError
 
-        Returns:
-            Returns a DataSource object based on the data_source protobuf
-        """
-
-        if isinstance(cls.options, FileOptions):
-            data_source = cls(file_options=data_source_proto.options,)
-        if isinstance(cls.options, BigQueryOptions):
-            data_source = cls(bigquery_options=data_source_proto.options,)
-        if isinstance(cls.options, KafkaOptions):
-            data_source = cls(kafka_options=data_source_proto.options,)
-        if isinstance(cls.options, KinesisOptions):
-            data_source = cls(kinesis_options=data_source_proto.options,)
-        else:
-            raise TypeError(
-                "DataSource.from_proto: Provided DataSource option is invalid. Only FileOptions, BigQueryOptions, KafkaOptions and KinesisOptions are supported currently."
-            )
+
+class FileSource(DataSource):
+    def __init__(
+        self,
+        field_mapping,
+        timestamp_column,
+        file_format,
+        file_url,
+        date_partition_column="",
```

Comment:
Why is this set to `""`?

Reply:
It is an optional field, added typings to the respective native classes.